### PR TITLE
Restore packages if needed

### PR DESCRIPTION
**Pull request by jrieken**
This PR makes sure that OmniSharp automatically restores packages (via ```dnu restore``` or ```kpm restore```) if needed. The flow goes like this:

1. OmniSharp receives a _dependencies_-message from DTH which flags ```unresolved``` dependencies
2. OmniSharp runs ```dnu restore``` or ```kpm restore``` (depending on the target runtime)
3. Upon completion OmniSharp sends DTH _restoreComplete_-messages (one per project)
4. DTH updates project references etc which in return makes OmniSharp update the Roslyn project
5. Roslyn emit a project modification event and forwards that to the editor
6. The editor ask request a new code check etc

### CODE REVIEW

File: src/OmniSharp/AspNet5/AspNet5ProjectSystem.cs [hunk 1]
```diff
@@ -266,6 +266,18 @@ public void Initalize()
                             frameworkProject.RawReferences.Remove(pair.Key);
                         }
                     }
+                    else if (m.MessageType == "Dependencies")
+                    {
+                        var val = m.Payload.ToObject<DependenciesMessage>();
+                        var unreolvedDependencies = val.Dependencies.Values
+                            .Where(dep => dep.Type == "Unresolved");
+
+                        if (unreolvedDependencies.Any())
```
Comment (nosami) [inline src/OmniSharp/AspNet5/AspNet5ProjectSystem.cs]: typo :)

**Comment by nosami** [comment]
What triggers DTH to send a dependencies message? How do I test this?

**Comment by jrieken** [comment]
@nosami - the easiest is to have project.json file with a broken dependency (like below). DTH always sends this message when a project is (re)initialized and dependencies that couldn't be resolved have a special type. 

```
{
	"dependencies": {},
	"commands": {
        "run": "run"
    },
	"frameworks": {
		"aspnet50": {},
		"aspnetcore50": {
			"dependencies": {
				"System.Consola": "4.0.0-beta-*"
			}
		}
	}
}
```

**Comment by jrieken** [comment]
and all of it makes sense when you fire up the ```/filesChanged```-route to tell OmniSharp that project.json has changed. That will also trigger this mechanism.

**Comment by nosami** [comment]
Yeah... I need to get that added to the editors. The code looks good. Looking forward to trying this out :+1:

**Comment by jrieken** [comment]
I am making some more changes as dnu restore won't support recursive restore much longer. Therefore, I going back to a model where I run restore on a project by project basis (instead of looking for global.json)

**Comment by nosami** [comment]
OK... Just let me know when you're happy to merge it.

**Comment by jrieken** [comment]
yeah - made the whole things simpler. it now one method that (in a task) starts and waits for the process. also locks simpler.

**Comment by davidfowl** [comment]
Sweet! Tested it and it works great!

**Pull request merged**